src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -341,8 +341,12 @@ private function addSessionSection(ArrayNodeDefinition $rootNode)
                         ->scalarNode('handler_id')->defaultValue('session.handler.native_file')->end()
                         ->scalarNode('name')
                             ->validate()
-                                ->ifTrue(function ($v) { return false !== strpos($v, '.'); })
-                                ->thenInvalid('Session name can not contain character "."')
+                                ->ifTrue(function ($v) {
+                                    parse_str($v, $parsed);
+
+                                    return implode('&', array_keys($parsed)) !== (string) $v;
+                                })
+                                ->thenInvalid('Session name %s contains illegal character(s)')
```

Comment:
missing double quotes around %s
about the implode, this means `&` is valid in a session name? how strange is that :)
but OK, that's because cookie are not encoded using URL queries.
This also means we should exclude `;`, isn't it?

Reply:
Appveryor fails for some reason on `'` char even though it's valid... I've just removed it from test case